src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
I think I agree with @arielb1 and @Aatch. As for incr. comp., I would imagine that "debug" and "release" artifacts are indeed just separate. (As opposed to partially reusable.) We could in principle detect whether things like overflow were relevant I guess etc.